probeinterface/io.py

```diff
@@ -99,6 +99,292 @@ def write_probeinterface(file, probe_or_probegroup):
+    df = pd.read_csv(folder.joinpath('probes.tsv'), sep='\t', header=0,
+                     keep_default_na=False, dtype=str)
+    df.replace(to_replace={'n/a': None}, inplace=True)
+
+    if 'probe_id' not in df:
+        raise ValueError('probes.tsv file does not contain probe_id column')
+    for row_idx, row in df.iterrows():
+        probe = Probe()
+        probe.annotate(**dict(row.items()))
+
+        # for string based annotations use '' instead of None as default
+        for string_annotation in ['name', 'manufacturer']:
+            if probe.annotations.get(string_annotation, None) is None:
+                probe.annotations[string_annotation] = ''
+
+        probes[str(probe.annotations['probe_id'])] = probe
+
+    # Step 2: READING CONTACTS.TSV
+    df = pd.read_csv(folder.joinpath('contacts.tsv'), sep='\t', header=0,
+                     keep_default_na=False, dtype=str)
+    df.replace(to_replace={'n/a': None}, inplace=True)
+
+    if 'probe_id' not in df:
+        raise ValueError('probes.tsv file does not contain probe_id column')
+    if 'contact_id' not in df:
+        raise ValueError('contacts.tsv file does not contain contact_id column')
+
+    for probe_id in df['probe_id'].unique():
+        df_probe = df[df['probe_id'] == probe_id]
+        probe = probes[str(probe_id)]
+
+        probe.electrode_ids = df_probe['contact_id'].values.astype('U')
+        if 'contact_shape' in df_probe:
+            probe.electrode_shapes = df_probe['contact_shape'].values
+
+        dimensions = []
+        for dim in ['x', 'y', 'z']:
+            if dim in df_probe:
+                dimensions.append(dim)
+
+        coordinates = df_probe[dimensions].values
+        probe.electrode_positions = coordinates.astype(float)
+        probe.ndim = len(dimensions)
+
+        # extract physical units used
+        if 'xyz_units' in df_probe:
+            if len(df['xyz_units'].unique()) == 1:
+                probe.si_units = df['xyz_units'][0]
+            else:
+                raise ValueError(f'Probe {probe_id} uses different units for '
+                                 f'contact coordinates. Can not be represented '
+                                 f'by probeinterface.')
+
+        if 'device_channel_indices' in df_probe:
+            channel_indices = df_probe['device_channel_indices'].values
+            if any([ind is not None for ind in channel_indices]):
+                probe.set_device_channel_indices(channel_indices)
+
+        if 'shank_id' in df_probe:
+            shank_ids = df_probe['shank_id'].values
+            probe.set_shank_ids(shank_ids)
+
+        used_columns = ['probe_id', 'contact_id', 'contact_shape',
+                        'x', 'y', 'z', 'xyz_units', 'shank_id',
+                        'device_channel_indices']
+        df_others = df_probe.drop(used_columns, axis=1, errors='ignore')
+        for col_name in df_others.columns:
+            probe.annotate(**{col_name: df_probe[col_name].values})
```

Comment:
Maybe we should also introduce an **array_annotations** as we do in neo for this, no ?

Reply:
Yes, but this should be a separate PR.

Reply:
of course.